### PR TITLE
[vernac] Remove `Qed exporting` syntax.

### PR DESCRIPTION
**Pull request by ejgallego**
This is an RFC to start to deprecate functionality breaking opaqueness of Qed.

We don't gain anything from the kernel yet as transparent constants _do_ require the `side_eff` exporting machinery, but that would be the next step.

### CODE REVIEW

File: doc/refman/RefMan-ltac.tex
Comment (JasonGross) [inline doc/refman/RefMan-ltac.tex]: Grammar nit: "Such an auxiliary lemma is" or "Such auxiliary lemmas are"

File: doc/tutorial/Tutorial.tex
Comment (JasonGross) [inline doc/tutorial/Tutorial.tex]: This sentence is wrong; `Qed` never takes a name argument.

File: man/gallina.1
Comment (JasonGross) [inline man/gallina.1]: Should this include `Save ident.`?

File: tools/gallina-syntax.el
Comment (JasonGross) [inline tools/gallina-syntax.el]: I think this is wrong, since you can still end something with `Save ident`

**Review by JasonGross — DISMISSED** [review]

**Comment by ejgallego** [comment]
Thanks @JasonGross , indeed the commit relation to save depend on #642 , I'll try to sync both.

**Comment by maximedenes** [comment]
What do people think of this? @gares @herbelin @mattam82 ?

**Comment by Zimmi48** [comment]
@ejgallego Since this is only a small step towards a goal, why include this in 8.7? Why not instead make a pull request only deprecating this functionality in 8.7 and merge this PR removing the functionality in 8.8?

**Comment by ejgallego** [comment]
I don't particularly mind either option @Zimmi48

**Comment by maximedenes** [comment]
I would be in favor of deprecating rather than removing, as a first step. @ejgallego, can you provide a PR that does the deprecation?

**Comment by ejgallego** [comment]
How does the deprecation work here? Just an entry in the manual?

**Comment by maximedenes** [comment]
How about an entry in the manual and a warning when used?

**Comment by ejgallego** [comment]
I guess we may just add a note in the manual stating that this feature is experimental, we may even keep it once we can agree on the STM roadmap for 8.8.

**Comment by maximedenes** [comment]
I think any change should be discussed with @gares, but I'm afraid he's away and busy until the freeze. If this is going to become a doc fix only, it can be integrated after freeze anyway.

**Comment by ejgallego** [comment]
@JasonGross dismissing the review as I think I solved your comments, thanks!

**Comment by ppedrot** [comment]
We should bring back this on the table. I'm also in favour of removing this feature. The question of a first phase of deprecation should be asked indeed, but I'm wondering whether people actually rely on this command, and if so, how deeply. You can always emulate Qed exporting with Defined + subsequent opaque redefinition.

**Comment by ejgallego** [comment]
Pierre Marie, IIRC nobody has ever used this so I think we could just safely remove it.

**Comment by ejgallego** [comment]
Anyways for extra safety, we could just deprecate it in 8.7 and remove in `master`.

**Comment by ppedrot** [comment]
@ejgallego LGTM.

**Comment by ejgallego** [comment]
Ok I will prepare a PR for 8.7.

**Comment by ejgallego** [comment]
But before I proceed, we need @gares to approve the plan.

**Comment by ppedrot** [comment]
BTW, according to the commit that introduced it (4a53151), this is an explicit compatibility mode for 8.4. The command should have been flagged as deprecated a long time ago.

**Comment by Zimmi48** [comment]
Has the PR for deprecating this in 8.7 ever been made? Apparently @ejgallego wanted some explicit approval from @gares. Do you approve this plan @gares?

**Comment by gares** [comment]
I've basically added it against my will... of course I want to get rid of it ;-)

**Comment by ejgallego** [comment]
I couldn't track any users so we should be safe here I think.

**Comment by Zimmi48** [comment]
So to me, there is no need to wait any longer. We can proceed to merge this PR as soon as a v8.7-only PR has been merged deprecating this (unless the deprecation has already been done and I forgot about it). But please do it, the rule is no removal without deprecation.

File: vernac/lemmas.ml
Comment (gares) [inline vernac/lemmas.ml]: Exports is now always `[]` and hence `check_exports` down below is dead code.

**Review by gares — CHANGES_REQUESTED** [review]

File: vernac/lemmas.ml
Comment (ejgallego) [reply vernac/lemmas.ml]: Well spotted, grazie; fixed.

**Comment by ejgallego** [comment]
Comments addressed, ready to merge.

**Comment by Zimmi48** [comment]
Maybe it's time to remove `[RFC]` from the title of the PR, then ;-)

**Pull request merged**
